### PR TITLE
[SYCL] Support target::host_task specialised accessor constructor using mode_target Tag.

### PR DESCRIPTION
**Pull request by mmoadeli**
- Support target::host_task specialised accessor constructor using mode_target Tag.
- Add minimal test.

### CODE REVIEW

File: sycl/include/sycl/accessor.hpp [hunk 1]
```diff
@@ -414,6 +436,28 @@ constexpr access::target deduceAccessTarget(access::target defaultTarget) {
...
+                   mode_target_tag_t<access::mode::write,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag1,
+                   mode_target_tag_t<access::mode::read_write,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag2,
+                   mode_target_tag_t<access::mode::read_write,
+                                     access::target::host_task>>::value) {
+    return access::target::host_task;
+  }
```
Comment (steffenlarsen) [inline sycl/include/sycl/accessor.hpp:459]: Eventually I would love to rewrite this as trait/function utilizing std::optional so we don't need to write out all combinations. That said, you're following the existing implementation so I will not ask it of you.

**Review by steffenlarsen — APPROVED** [review]
Looks good!

**Comment by steffenlarsen** [comment]
@mmoadeli - There were some testing issues around the time you opened this PR. Could you please push a merge commit to rerun tests after it was fixed?

**Comment by mmoadeli** [comment]
> @mmoadeli - There were some testing issues around the time you opened this PR. Could you please push a merge commit to rerun tests after it was fixed?

**Pull request merged**